### PR TITLE
extracting data from seeders

### PR DESCRIPTION
**Pull request by CamiloGilFranco**
## Summary
- Modification is added to the rating of the restaurants to have a data according to the ratings of the users
- Modification is added to the number of restaurant sales to be consistent with the order_details table
- Encryption is generated to user passwords generated by the seeder
- Functionality is added to the users seeder to create a text file with the credentials of the users created by the seeders
- An update is made to the database, removing the obligation of a rating when creating a restaurant
- The entity relationship diagram is updated

### CODE REVIEW

File: seeders/reviewsSeeder.ts [hunk 1]
```diff
@@ -30,10 +35,26 @@ const seedReviews = async (prisma: PrismaClient): Promise<void> => {
...
+      data: {
+        rating: ratingAverage,
+      },
+    });
   }
 
   console.log("reviews created!");
 };
 
+/* seedReviews(prisma); */
```
Comment (ht1204) [inline seeders/reviewsSeeder.ts:58]: what is this commented code doing?

File: seeders/reviewsSeeder.ts [hunk 2]
```diff
@@ -30,10 +35,26 @@ const seedReviews = async (prisma: PrismaClient): Promise<void> => {
           },
         },
       });
+
+      dataQuantity++;
+      ratingSum += singleRating;
     }
+
+    const ratingAverage = ratingSum / dataQuantity;
```
Comment (ht1204) [inline seeders/reviewsSeeder.ts:43]: `const ratingAverage = ratingSum / dataQuantity;`
```js
await prisma.restaurants.update({
      where: {
        id_restaurant: restaurantID,
      },
      data: {
        rating,
      },
    });
  }
```

File: seeders/order_detailsSeeder.ts [hunk 3]
```diff
@@ -67,6 +66,29 @@ const seedOrder_details = async (prisma: PrismaClient): Promise<void> => {
   }
 
   console.log("order prices updated!");
+
+  const restaurants = await prisma.restaurants.findMany();
+
+  for (const restaurant of restaurants) {
+    const restaurantId = restaurant.id_restaurant;
```
Comment (ht1204) [inline seeders/order_detailsSeeder.ts:73]: `const { id_restaurant }= restaurant;`

File: seeders/order_detailsSeeder.ts [hunk 4]
```diff
@@ -67,6 +66,29 @@ const seedOrder_details = async (prisma: PrismaClient): Promise<void> => {
...
+
+    const shopping = await prisma.order_details.findMany({
+      where: {
+        restaurantsId_restaurant: restaurantId,
```
Comment (ht1204) [inline seeders/order_detailsSeeder.ts:77]: `restaurantsId_restaurant: id_restaurant`

File: seeders/order_detailsSeeder.ts [hunk 5]
```diff
@@ -67,6 +66,29 @@ const seedOrder_details = async (prisma: PrismaClient): Promise<void> => {
...
+      },
+    });
+
+    await prisma.restaurants.update({
+      where: {
+        id_restaurant: restaurantId,
```
Comment (ht1204) [inline seeders/order_detailsSeeder.ts:83]: `id_restaurant,`

File: seeders/order_detailsSeeder.ts [hunk 6]
```diff
@@ -67,6 +66,29 @@ const seedOrder_details = async (prisma: PrismaClient): Promise<void> => {
...
+
+  for (const restaurant of restaurants) {
+    const restaurantId = restaurant.id_restaurant;
+
+    const shopping = await prisma.order_details.findMany({
+      where: {
+        restaurantsId_restaurant: restaurantId,
+      },
+    });
+
```
Comment (ht1204) [inline seeders/order_detailsSeeder.ts:80]: `const {length: number_of_sales} = shopping;`

File: seeders/order_detailsSeeder.ts [hunk 7]
```diff
@@ -67,6 +66,29 @@ const seedOrder_details = async (prisma: PrismaClient): Promise<void> => {
...
+    await prisma.restaurants.update({
+      where: {
+        id_restaurant: restaurantId,
+      },
+      data: {
+        number_of_sales: shopping.length,
```
Comment (ht1204) [inline seeders/order_detailsSeeder.ts:86]: `number_of_sales,`

File: seeders/reviewsSeeder.ts [hunk 8]
```diff
@@ -11,16 +11,21 @@ const seedReviews = async (prisma: PrismaClient): Promise<void> => {
 
   for (const restaurant of restaurants) {
     const numberOfReviews = random(0, 20);
+    const restaurantID = restaurant.id_restaurant;
+    let ratingSum = 0;
+    let dataQuantity = 0;
 
     for (let i = 0; i < numberOfReviews; i++) {
+      const singleRating = random(1, 5);
```
Comment (ht1204) [inline seeders/reviewsSeeder.ts:19]: `const rating = random(1, 5);`

File: seeders/reviewsSeeder.ts [hunk 9]
```diff
@@ -11,16 +11,21 @@ const seedReviews = async (prisma: PrismaClient): Promise<void> => {
...
+
       await prisma.reviews.create({
         data: {
           title: faker.lorem.words(random(1, 5)),
-          rating: random(1, 5),
+          rating: singleRating,
```
Comment (ht1204) [inline seeders/reviewsSeeder.ts:24]: `rating,`

File: seeders/reviewsSeeder.ts [hunk 10]
```diff
@@ -11,16 +11,21 @@ const seedReviews = async (prisma: PrismaClient): Promise<void> => {
 
   for (const restaurant of restaurants) {
     const numberOfReviews = random(0, 20);
+    const restaurantID = restaurant.id_restaurant;
```
Comment (ht1204) [inline seeders/reviewsSeeder.ts:14]: `const { id_restaurant }= restaurant;`

File: seeders/reviewsSeeder.ts [hunk 11]
```diff
@@ -11,16 +11,21 @@ const seedReviews = async (prisma: PrismaClient): Promise<void> => {
...
       await prisma.reviews.create({
         data: {
           title: faker.lorem.words(random(1, 5)),
-          rating: random(1, 5),
+          rating: singleRating,
           comment: faker.lorem.sentence(),
           restaurants: {
             connect: {
-              id_restaurant: restaurant.id_restaurant,
+              id_restaurant: restaurantID,
```
Comment (ht1204) [inline seeders/reviewsSeeder.ts:28]: `id_restaurant,`

File: seeders/reviewsSeeder.ts [hunk 12]
```diff
@@ -11,16 +11,21 @@ const seedReviews = async (prisma: PrismaClient): Promise<void> => {
 
   for (const restaurant of restaurants) {
     const numberOfReviews = random(0, 20);
+    const restaurantID = restaurant.id_restaurant;
+    let ratingSum = 0;
+    let dataQuantity = 0;
 
     for (let i = 0; i < numberOfReviews; i++) {
+      const singleRating = random(1, 5);
+
```
Comment (ht1204) [inline seeders/reviewsSeeder.ts:20]: `const { user_id } = usersList[random(0, usersList.length - 1)];`

File: seeders/usersSeeder.ts [hunk 13]
```diff
@@ -13,16 +15,24 @@ const seedUsers = async (prisma: PrismaClient): Promise<void> => {
     appAdmin: "appAdmin",
   };
 
+  if (existsSync("./assets/usersData.txt")) {
+    unlinkSync("./assets/usersData.txt");
```
Comment (ht1204) [inline seeders/usersSeeder.ts:19]: interesting...

File: seeders/usersSeeder.ts [hunk 14]
```diff
@@ -13,16 +15,24 @@ const seedUsers = async (prisma: PrismaClient): Promise<void> => {
...
+
   const restaurants = await prisma.restaurants.findMany();
 
   for (const restaurant of restaurants) {
     const numberOfAdmins = random(1, 5);
 
     for (let i = 0; i < numberOfAdmins; i++) {
+      const userEmail = faker.internet.email();
+      const userPassword = faker.internet.password();
+      const cryptPassword = await bcrypt.hash(userPassword, 10);
```
Comment (ht1204) [inline seeders/usersSeeder.ts:30]:
```js
  const email = faker.internet.email();
  const userPassword = faker.internet.password();
  const password = await bcrypt.hash(userPassword, 10);
  const name: = faker.name.firstName();
  const last_name = faker.name.lastName(),
  const document_type = documentTypes[random(0, documentTypes.length - 1)],
```

File: seeders/usersSeeder.ts [hunk 15]
```diff
@@ -13,16 +15,24 @@ const seedUsers = async (prisma: PrismaClient): Promise<void> => {
...
+
       await prisma.users.create({
         data: {
-          email: faker.internet.email(),
-          password: faker.internet.password(),
+          email: userEmail,
+          password: cryptPassword,
           name: faker.name.firstName(),
           last_name: faker.name.lastName(),
           document_type: documentTypes[random(0, documentTypes.length - 1)],
```
Comment (ht1204) [inline seeders/usersSeeder.ts:38]:
```js
data : {
 email,
 password,
 name,
 last_name,
 document_type,
}
```

File: seeders/usersSeeder.ts [hunk 16]
```diff
@@ -44,13 +54,22 @@ const seedUsers = async (prisma: PrismaClient): Promise<void> => {
           },
         },
       });
+
+      appendFileSync(
+        "./assets/usersData.txt",
+        `Rol: restaurantAdmin --- Email: ${userEmail} --- Password: ${userPassword} --- AdminOf: ${restaurant.restaurant_name}\n`
```
Comment (ht1204) [inline seeders/usersSeeder.ts:60]: interesting...

File: seeders/usersSeeder.ts [hunk 17]
```diff
@@ -44,13 +54,22 @@ const seedUsers = async (prisma: PrismaClient): Promise<void> => {
...
+      );
     }
   }
 
   for (let i = 0; i < 100; i++) {
+    const userEmail = faker.internet.email();
+    const userPassword = faker.internet.password();
+    const cryptPassword = await bcrypt.hash(userPassword, 10);
```
Comment (ht1204) [inline seeders/usersSeeder.ts:68]: same prop assignment here would be an enhancement.

File: seeders/usersSeeder.ts [hunk 18]
```diff
@@ -64,12 +83,21 @@ const seedUsers = async (prisma: PrismaClient): Promise<void> => {
...
+    appendFileSync(
+      "./assets/usersData.txt",
+      `Rol: user --- Email: ${userEmail} --- Password: ${userPassword}\n`
+    );
   }
 
   for (let i = 0; i < random(1, 10); i++) {
+    const userEmail = faker.internet.email();
+    const userPassword = faker.internet.password();
+    const cryptPassword = await bcrypt.hash(userPassword, 10);
```
Comment (ht1204) [inline seeders/usersSeeder.ts:96]: and here

File: seeders/order_detailsSeeder.ts (hunk 6 above)
Comment (CamiloGilFranco) [reply seeders/order_detailsSeeder.ts:80]: Doesn't work

**Pull request merged**